pilz_extensions/test/integrationtest_pilz_extensions_demo.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2019 Pilz GmbH & Co. KG
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import roslaunch
+import rospkg
+import rospy
+
+PACKAGE_NAME = 'pilz_extensions'
+
+## Test the pilz_extensions demo
+#
+# This test exists mainly for full coverage
```

Comment:
I think at least you need some more explaining to to here.

Reply:
Maybe this is wrong?
I applied 
```
diff --git a/pilz_extensions/CMakeLists.txt b/pilz_extensions/CMakeLists.txt
index 2ae434f..5ae605e 100644
--- a/pilz_extensions/CMakeLists.txt
+++ b/pilz_extensions/CMakeLists.txt
@@ -43,11 +43,11 @@ include_directories(
  ${catkin_INCLUDE_DIRS}
 )
 
-add_executable(${PROJECT_NAME}_demo_node src/pilz_extensions_demo_node.cpp)
+# add_executable(${PROJECT_NAME}_demo_node src/pilz_extensions_demo_node.cpp)
 
-target_link_libraries(${PROJECT_NAME}_demo_node
-   ${catkin_LIBRARIES}
- )
+# target_link_libraries(${PROJECT_NAME}_demo_node
+#    ${catkin_LIBRARIES}
+#  )
 
 #############
 ## Install ##
@@ -60,11 +60,11 @@ install(DIRECTORY include/${PROJECT_NAME}/
   PATTERN ".svn" EXCLUDE
 )
 
-install(TARGETS ${PROJECT_NAME}_demo_node
-   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-)
+# install(TARGETS ${PROJECT_NAME}_demo_node
+#    ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+#    LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+#    RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+# )
 
 
 #############
@@ -87,9 +87,9 @@ if(CATKIN_ENABLE_TESTING)
      ${catkin_LIBRARIES})
 
   # Integrationtest of pilz_extensions_demo_node
-  add_rostest(test/integrationtest_pilz_extensions_demo.test)
+  # add_rostest(test/integrationtest_pilz_extensions_demo.test)
 
-  roslint_python(test/integrationtest_pilz_extensions_demo.py)
+  # roslint_python(test/integrationtest_pilz_extensions_demo.py)
 
   # to run: catkin_make -DENABLE_COVERAGE_TESTING=ON package_name_coverage
   if(ENABLE_COVERAGE_TESTING)
```
and 
`catkin_make -DENABLE_COVERAGE_TESTING=ON && catkin_make -DENABLE_COVERAGE_TESTING=ON pilz_extensions_coverage` gives me 100%. However I need to run it twice, which is strange...

Reply:
I don't get your point. You are commenting out the demo node? For what reason?

Reply:
I reformulated the test documentation a bit.

Reply:
Let's continue discussion here. Changes will be pushed to #222